### PR TITLE
feat(helm): add support to p2p

### PR DESCRIPTION
**Pull request by idan-starkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/2212)

### CODE REVIEW

File: deployments/helm/templates/deployment.yaml [hunk 1]
```diff
@@ -57,11 +62,29 @@ spec:
             cpu: {{ .Values.deployment.resources.requests.cpu | quote}}
             memory: {{ .Values.deployment.resources.requests.memory }}
         {{- if not .Values.backup.enabled }}
+        {{- with .Values.deployment.env }}
+        env:
+          {{- toYaml . | nindent 12 }}
```
Comment (eranreshef-starkware) [inline deployments/helm/templates/deployment.yaml:67]: are you sure the indentation here needs to be 12?

File: deployments/helm/templates/deployment.yaml [hunk 2]
```diff
@@ -57,11 +62,29 @@ spec:
...
+        {{- end }}
         args:
           - --config_file 
           - /app/config/presets/{{ .Values.starknet.preset }}
           - --base_layer.node_url 
           - {{ .Values.base_layer_node_url }}
+          {{- if .Values.p2p.enabled }}
+          - --network.tcp_port
+          - {{ default "10000" .Values.p2p.config.networkTcpPort | quote }}
```
Comment (eranreshef-starkware) [inline deployments/helm/templates/deployment.yaml:76]: not sure its a good idea to give default values inside the templates. if you specify the value in the default `values.yaml` you get the same result and its more explicit and readable.

**Review by eranreshef-starkware — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/2212#-:-O1QaU7XAfyPKIZ0VS74:bnfp4nl)* status: 2 of 6 files reviewed, all discussions resolved

**Review by eranreshef-starkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/2212)* status: 1 of 6 files reviewed, 2 unresolved discussions (waiting on @idan-starkware)

___
*[`deployments/helm/values.yaml` line 11 at r5](https://reviewable.io/reviews/starkware-libs/papyrus/2212#-O1zgP0NEeCx6A_Kvw9w:-O1zgP0NEeCx6A_Kvw9x:b-8cbk8p) ([raw file](https://github.com/starkware-libs/papyrus/blob/1dc1db8ada64fb4c93db0653ff47370c87aa138d/deployments/helm/values.yaml#L11)):*
> ```YAML
> 
> # Ethereum node URL. A value for this variable is mandatory.
> base_layer_node_url: https://base_url.example.com
> ```

this config field shouldn't get a default value because there is no default value that will work globally.
___
*[`deployments/helm/values.yaml` line 96 at r5](https://reviewable.io/reviews/starkware-libs/papyrus/2212#-O1zg4cp9ASu_C6-0pgS:-O1zg4cp9ASu_C6-0pgT:bie6sro) ([raw file](https://github.com/starkware-libs/papyrus/blob/1dc1db8ada64fb4c93db0653ff47370c87aa138d/deployments/helm/values.yaml#L96)):*
> ```YAML
> pvc:
>   # Recommended size is at least 512Gi.
>   size: 1Gi
> ```

as discussed verbally yesterday, either set the recommended size or don't set a size at all.

**Review by idan-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/2212)* status: 0 of 6 files reviewed, 2 unresolved discussions (waiting on @eranreshef-starkware and @idan-starkware)

___
*[`deployments/helm/values.yaml` line 11 at r5](https://reviewable.io/reviews/starkware-libs/papyrus/2212#-O1zgP0NEeCx6A_Kvw9w:-O2-J3kJ-k3pnIMDAv97:b-896fix) ([raw file](https://github.com/starkware-libs/papyrus/blob/1dc1db8ada64fb4c93db0653ff47370c87aa138d/deployments/helm/values.yaml#L11)):*
<details><summary><i>Previously, eranreshef-starkware (Eran Reshef) wrote…</i></summary><blockquote>

this config field shouldn't get a default value because there is no default value that will work globally.
</blockquote></details>

Done.
___
*[`deployments/helm/values.yaml` line 96 at r5](https://reviewable.io/reviews/starkware-libs/papyrus/2212#-O1zg4cp9ASu_C6-0pgS:-O2-J3zFAtXnzEw3naAd:b-896fix) ([raw file](https://github.com/starkware-libs/papyrus/blob/1dc1db8ada64fb4c93db0653ff47370c87aa138d/deployments/helm/values.yaml#L96)):*
<details><summary><i>Previously, eranreshef-starkware (Eran Reshef) wrote…</i></summary><blockquote>

as discussed verbally yesterday, either set the recommended size or don't set a size at all.
</blockquote></details>

Done.

**Review by eranreshef-starkware — APPROVED** [review]
> Reviewed 4 of 6 files at r1, 1 of 2 files at r4, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/2212)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @idan-starkware)

**Pull request merged**